pkg/deploy/dev-workspace-config/dev_workspace_config.go

```diff
@@ -0,0 +1,113 @@
+	chev2 "github.com/eclipse-che/che-operator/api/v2"
+	"github.com/eclipse-che/che-operator/pkg/common/chetypes"
+	"github.com/eclipse-che/che-operator/pkg/common/constants"
+	"github.com/eclipse-che/che-operator/pkg/common/utils"
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	devWorkspaceConfigName = "devworkspace-config"
+)
+
+type DevWorkspaceConfigReconciler struct {
+	deploy.Reconcilable
+}
+
+func NewDevWorkspaceConfigReconciler() *DevWorkspaceConfigReconciler {
+	return &DevWorkspaceConfigReconciler{}
+}
+
+func (d *DevWorkspaceConfigReconciler) Reconcile(ctx *chetypes.DeployContext) (reconcile.Result, bool, error) {
+	dwoc := &controllerv1alpha1.DevWorkspaceOperatorConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      devWorkspaceConfigName,
+			Namespace: ctx.CheCluster.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DevWorkspaceOperatorConfig",
+			APIVersion: controllerv1alpha1.GroupVersion.String(),
+		},
+	}
+
+	if _, err := deploy.GetNamespacedObject(ctx, devWorkspaceConfigName, dwoc); err != nil {
+		return reconcile.Result{}, false, err
+	}
+
+	if dwoc.Config == nil {
+		dwoc.Config = &controllerv1alpha1.OperatorConfiguration{}
+	}
+	updateOperatorConfig(ctx.CheCluster.Spec.DevEnvironments.Storage, dwoc.Config)
+
+	done, err := deploy.Sync(ctx, dwoc)
+	if !done {
+		return reconcile.Result{}, false, err
+	}
+
+	return reconcile.Result{}, true, nil
+}
+
+func (d *DevWorkspaceConfigReconciler) Finalize(ctx *chetypes.DeployContext) bool {
+	return true
+}
+
+func updateOperatorConfig(storage chev2.WorkspaceStorage, operatorConfig *controllerv1alpha1.OperatorConfiguration) {
+	var pvc *chev2.PVC
+
+	pvcStrategy := utils.GetValue(storage.PvcStrategy, constants.DefaultPvcStorageStrategy)
+	switch pvcStrategy {
+	case constants.CommonPVCStorageStrategy:
+		fallthrough
+	case constants.PerUserPVCStorageStrategy:
+		if storage.PerUserStrategyPvcConfig != nil {
+			pvc = storage.PerUserStrategyPvcConfig
+		}
+	case constants.PerWorkspacePVCStorageStrategy:
+		if storage.PerWorkspaceStrategyPvcConfig != nil {
+			pvc = storage.PerWorkspaceStrategyPvcConfig
+		}
+	}
+
+	if pvc != nil {
+		if operatorConfig.Workspace == nil {
+			operatorConfig.Workspace = &controllerv1alpha1.WorkspaceConfig{}
+		}
+		updateWorkspaceConfig(pvc, pvcStrategy == constants.PerWorkspacePVCStorageStrategy, operatorConfig.Workspace)
+	}
+}
+
+func updateWorkspaceConfig(pvc *chev2.PVC, isPerWorkspacePVCStorageStrategy bool, workspaceConfig *controllerv1alpha1.WorkspaceConfig) {
+	if pvc.StorageClass != "" {
+		workspaceConfig.StorageClassName = &pvc.StorageClass
+	}
+
+	if pvc.ClaimSize != "" {
+		if workspaceConfig.DefaultStorageSize == nil {
+			workspaceConfig.DefaultStorageSize = &controllerv1alpha1.StorageSizes{}
+		}
+
+		pvcSize := resource.MustParse(pvc.ClaimSize)
```

Comment:
@tolusha thank you so much for the refactor, things are looking much cleaner now.

The only issue I've found is that, since the `pvc.claimSize` is a string in the CRD (and not a resource.Quantity), it's possible for `resource.MustParse(pvc.ClaimSize)` to fail and cause a panic. `resource.ParseQuantity()` should be used instead.

Want me to submit a fixup for this?

Reply:
I submitted the fix up. In the process I had rebased the branch which accidentally dragged some extra commits onto this PR, sorry about that. I can remove those from the PR if desired.

Reply:
Everything looks fine. Merge if needed.